service/integration/open_pbs/README.rst

```diff
@@ -28,57 +36,250 @@ correct the power limit setting if it was not restored as part of the previous
 
 - OpenPBS
 - geopmdpy (and libgeopmd) on nodes where the power limit feature is needed
 - This feature requires platform vendor HW support. In particular, it requires
-  HW support for the MSR::PLATFORM_POWER_LIMIT MSRs (PL1_POWER_LIMIT,
-  PL1_TIME_WINDOW, PL1_CLAMP_ENABLE, PL1_LIMIT_ENABLE), and the
-  MSR::PLATFORM_ENERGY_STATUS MSRs.
+  HW support for the ``MSR::PLATFORM_POWER_LIMIT`` MSRs (``PL1_POWER_LIMIT``,
+  ``PL1_TIME_WINDOW``, ``PL1_CLAMP_ENABLE``, ``PL1_LIMIT_ENABLE``), and the
+  ``MSR::PLATFORM_ENERGY_STATUS`` MSRs.
 
 Installation
 ------------
-
 Prior to being able to use the power limit feature, the
-``geopm-node-power-limit`` resource and the power limit hook need to be
-installed on the OpenPBS server.
+``geopm-node-power-limit`` and ``geopm-job-power-limit`` resources and the
+power limit hook need to be installed on the OpenPBS server.
 
-The resource can be created with the following command:
+The resources can be created with the following commands:
 
 ::
 
-   qmgr -c "create resource geopm-node-power-limit type=long"
+   qmgr -c "create resource geopm-node-power-limit type=float"
+   qmgr -c "create resource geopm-min-node-power-limit type=float,flag=r"
+   qmgr -c "create resource geopm-max-node-power-limit type=float,flag=r"
+   qmgr -c "create resource geopm-job-power-limit type=float,flag=q"
+   qmgr -c "create resource geopm-max-slowdown type=float"
+   qmgr -c "create resource geopm-job-type type=string"
 
-The hook is available in the ``service/integration/open_pbs`` directory
-(``geopm_power_limit.py``) and it can be installed with the following
-commands:
+To ensure that the job power limit is treated as a consumable resource by the
+PBS scheduler, the resource must also be declared in the scheduler's
+configuration file (located at ``$PBS_HOME/sched_priv/sched_config``). That is,
+the line beginning with ``resources:`` should include an entry for
+``geopm-job-power-limit``. The other resources mentioned in this guide do not
+need to be added to that configuration file. For example, the following
+commands add the GEOPM resource to an existing scheduler configuration:
+
+::
+
+    source /etc/pbs.conf
+    sed -i -e 's/^resources: "\([^"]\+\)"$/resources: "\1, geopm-job-power-limit"/g' "${PBS_HOME}/sched_priv/sched_config"
+
+Restart the PBS scheduler for the configuration changes to take effect. For
+example, one way is to use systemctl or run ``/etc/init.d/pbs restart`` where
+the PBS scheduler is running.
+
+The hook to enforce node power caps is available in the GEOPM
+``service/integration/open_pbs`` directory (``geopm_power_limit.py``) and
+can be installed with the following commands:
 
 ::
 
    qmgr -c "create hook geopm_power_limit"
    qmgr -c "import hook geopm_power_limit application/x-python default geopm_power_limit.py"
-   qmgr -c "set hook geopm_power_limit event='execjob_prologue,execjob_epilogue'"
+   qmgr -c "set hook geopm_power_limit event='execjob_prologue,execjob_epilogue,queuejob'"
```

Comment:
runjob seems to be missing

Reply:
It's erroneously mentioned in other parts of the docs. I'll remove those references from the upcoming revision.